### PR TITLE
WIP: New frontend

### PR DESCRIPTION
**Pull request by n8kim1**
Mega-PR for the new frontend

### CODE REVIEW

**Comment by acrantel** [comment]
Note: After we merge this PR, we'll need to regenerate the API because the backend in main is different from the backend in frontend2

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (acrantel) [inline .vscode/settings.json:1]: we should delete this before merging

**Review by lowtorola — APPROVED** [review]
LGTM!! 🎉

**Review by sri240 — APPROVED** [review]
flawless

**Pull request merged**